src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -199,10 +187,47 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
             // exact same options
             if (!_optionsCache.TryGetValue(sectionKey, out var result))
             {
+                var treeOptionsBuilder = _treeOptionsPool.Allocate();
+                var analyzerOptionsBuilder = _analyzerOptionsPool.Allocate();
+                var diagnosticBuilder = ArrayBuilder<Diagnostic>.GetInstance();
+
+                int sectionIndex = 0;
+                for (int analyzerConfigIndex = 0;
+                    analyzerConfigIndex < _analyzerConfigs.Length && sectionIndex < sectionKey.Count;
```

Comment:
In what circumstance is `sectionIndex < sectionKey.Count`, given the check below?

Reply:
The other check only exits the _inner_ `for` loop. This one exits the _outer_ for loop.

Reply:
ugh, mutlilevel break strikes again.

Reply:
Added a comment to improve clarity.